app/src/androidTest/java/org/simple/clinic/overdue/AppointmentRepositoryAndroidTest.kt

```diff
@@ -375,206 +420,228 @@ class AppointmentRepositoryAndroidTest {
-
-    val appointments = appointmentRepository.recordsWithSyncStatus(SyncStatus.PENDING).blockingGet()
-    assertThat(appointments).hasSize(1)
-    assertThat(appointments.first().remindOn).isNull()
-
-    appointmentRepository.setSyncStatus(from = SyncStatus.PENDING, to = SyncStatus.DONE).blockingAwait()
+    // given
+    val appointmentDate = LocalDate.parse("2018-01-01")
+    val appointmentScheduledAtTimestamp = Instant.now(clock)
+    appointmentRepository.schedule(
+        patientUuid = patientUuid,
+        appointmentUuid = appointmentUuid,
+        appointmentDate = appointmentDate,
+        appointmentType = Manual,
+        currentFacility = facility
+    ).blockingGet()
+    markAppointmentSyncStatusAsDone(appointmentUuid)
 
     testClock.advanceBy(Duration.ofHours(24))
 
-    val uuid = appointments[0].uuid
-    val reminderDate = LocalDate.now(clock).plusDays(10)
-    appointmentRepository.createReminder(uuid, reminderDate).blockingGet()
-
-    val updatedAppointments = appointmentRepository.recordsWithSyncStatus(SyncStatus.PENDING).blockingGet()
-    assertThat(updatedAppointments).hasSize(1)
-    updatedAppointments[0].apply {
-      assertThat(this.uuid).isEqualTo(uuid)
-      assertThat(this.remindOn).isEqualTo(reminderDate)
-      assertThat(this.agreedToVisit).isNull()
-      assertThat(this.syncStatus).isEqualTo(SyncStatus.PENDING)
-      assertThat(this.updatedAt).isNotEqualTo(timeOfSchedule)
-      assertThat(this.updatedAt).isEqualTo(Instant.now(clock))
+    val reminderDate = LocalDate.parse("2018-02-01")
+
+    // when
+    appointmentRepository.createReminder(appointmentUuid, reminderDate).blockingGet()
+
+    // then
+    val updatedAppointment = getAppointmentByUuid(appointmentUuid)
+    with(updatedAppointment) {
+      assertThat(remindOn).isEqualTo(reminderDate)
+      assertThat(agreedToVisit).isNull()
+      assertThat(syncStatus).isEqualTo(PENDING)
+      assertThat(createdAt).isEqualTo(appointmentScheduledAtTimestamp)
+      assertThat(updatedAt).isEqualTo(Instant.now(clock))
     }
   }
 
   @Test
-  fun when_marking_appointment_as_agreed_to_visit_reminder_for_30_days_should_be_set() {
-    val patientId = UUID.randomUUID()
-    val appointmentDate = LocalDate.now(clock)
-    val timeOfSchedule = Instant.now(clock)
-    appointmentRepository.schedule(patientId, appointmentDate, Manual, testData.qaFacility()).blockingGet()
-
-    val appointments = appointmentRepository.recordsWithSyncStatus(SyncStatus.PENDING).blockingGet()
-    assertThat(appointments).hasSize(1)
-    assertThat(appointments.first().remindOn).isNull()
-    assertThat(appointments.first().agreedToVisit).isNull()
-
-    val uuid = appointments[0].uuid
-    appointmentRepository.setSyncStatus(from = SyncStatus.PENDING, to = SyncStatus.DONE).blockingAwait()
+  fun when_marking_appointment_as_agreed_to_visit_reminder_for_1_month_should_be_set() {
```

Comment:
I intentionally changed it because I felt that it should be 1 month, and not 30 days. I was planning to make this change in a later PR.

My reasoning behind this was that if a patient was marked as "Agreed To Visit" on January 31st (last day of the month), I would expect the appointment to show up in overdue again on March 1st. However, if it's 30 days, then they'll start showing up in overdue again only on March 3rd (after 30 days).

Does this make sense?

Reply:
I clarified this with design right now (Pragati) and she feels that it should be "1 month", and not "30 days" as well.

Reply:
This was implemented like this because of the requirement. We were asked to schedule to 30 days later. Please get a confirmation from design team before making this change.
Let's keep this as it was before and once we get a confirmation, we can change this.
The name of the method is inconsistent with the actual behavior.

Reply:
I did get a confirmation.

Reply:
Then let's make the change in the real implementation and then only update the tests.
Otherwise, this is inconsistent.

Reply:
I got a confirmation from both Akshay and Pragati that we should use 1 calendar month and not 30 days. I will implement this in another PR.

Reply:
I m just saying that update the tests when you implement it.
And revert this change for now.

Reply:
I saw your earlier message now. Github did not refresh. I'll make this change.

Reply:
Reverted.